dlup/annotations_experimental.py

```diff
@@ -0,0 +1,1319 @@
+        "properties": {
+            "classification": {
+                "name": label,
+            },
+        },
+        "geometry": {},
+    }
+
+    if isinstance(geometry, Polygon):
+        # Construct the coordinates for the polygon
+        exterior = geometry.get_exterior()  # Get exterior coordinates
+        interiors = geometry.get_interiors()  # Get interior coordinates (holes)
+
+        # GeoJSON requires [ [x1, y1], [x2, y2], ... ] format
+        geojson["geometry"] = {
+            "type": "Polygon",
+            "coordinates": [[list(coord) for coord in exterior]]  # Exterior ring
+            + [[list(coord) for coord in interior] for interior in interiors],  # Interior rings (holes)
+        }
+
+    elif isinstance(geometry, Point):
+        # Construct the coordinates for the point
+        geojson["geometry"] = {
+            "type": "Point",
+            "coordinates": [geometry.x, geometry.y],
+        }
+    else:
+        raise ValueError(f"Unsupported geometry type {type(geometry)}")
+
+    if color is not None:
+        classification: dict[str, Any] = geojson["properties"]["classification"]
+        classification["color"] = color
+
+    return geojson
+
+
+def geojson_to_dlup(
+    coordinates: CoordinatesDict,
+    label: str,
+    color: Optional[tuple[int, int, int]] = None,
+    z_index: Optional[int] = None,
+) -> list[Polygon | Point]:
+    geom_type = coordinates.get("type", None)
+    if geom_type is None:
+        raise ValueError("No type found in coordinates.")
+    geom_type = geom_type.lower()
+
+    if geom_type in ["point", "multipoint"] and z_index is not None:
+        raise AnnotationError("z_index is not supported for point annotations.")
+
+    if geom_type == "point":
+        x, y = np.asarray(coordinates["coordinates"])
+        return [Point(*(x, y), label=label, color=color)]
+
+    if geom_type == "multipoint":
+        return [Point(*np.asarray(c).tolist(), label=label, color=color) for c in coordinates["coordinates"]]
+
+    if geom_type == "polygon":
+        _coordinates = coordinates["coordinates"]
+        polygon = Polygon(
+            np.asarray(_coordinates[0]), [np.asarray(hole) for hole in _coordinates[1:]], label=label, color=color
+        )
+        return [polygon]
+    if geom_type == "multipolygon":
+        output: list[Polygon | Point] = []
+        for polygon_coords in coordinates["coordinates"]:
+            exterior = np.asarray(polygon_coords[0])
+            interiors = [np.asarray(hole) for hole in polygon_coords[1:]]
+            output.append(Polygon(exterior, interiors, label=label, color=color))
+        return output
+
+    raise AnnotationError(f"Unsupported geom_type {geom_type}")
+
+
+class TagAttribute(NamedTuple):
+    label: str
+    color: Optional[tuple[int, int, int]]
+
+
+class SlideTag(NamedTuple):
+    attributes: Optional[list[TagAttribute]]
+    label: str
+    color: Optional[tuple[int, int, int]]
+
+
+class SlideAnnotations:
```

Comment:
If we want to have all image labels, annotations etc in this object and use that further in the pipeline it would be nice to also have a `from_database` something that parses labels from the database. Then you you can use the addition to combine the labels with the annotations if necessary in your usecase (in ahcore ofcourse)

Reply:
That's a good suggestion, but for that approach I would subclass and add this function since a database is not really standardized.